src/commcare_cloud/ansible/roles/formplayer/tasks/main.yml

```diff
@@ -54,24 +54,44 @@
     - formplayer_deploy
   with_items:
     - "{{ formplayer_build_dir }}"
-    - "{{ formplayer_release_dir }}/libs"
+    - "{{ formplayer_release_dir }}"
     - "{{ formplayer_data_dir }}"
 
 - name: Download formplayer release
   get_url:
     url: "{{ formplayer_build_url }}"
-    dest: "{{ formplayer_release_dir }}/libs/formplayer.jar"
+    dest: "{{ formplayer_release_dir }}/formplayer.jar"
     mode: 0744
     owner: "{{ cchq_user }}"
     group: "{{ cchq_user }}"
   tags:
     - formplayer_deploy
 
+- name: Check if the "current" directory exists
+  stat:
+    path: "{{ formplayer_build_dir }}/current"
+  register: current_stat
+  tags:
+    - formplayer_deploy
+
+- name: Copy config files from current release
+  become: yes
+  copy:
+    remote_src: yes
+    src: "{{ formplayer_build_dir }}/current/{{ item }}"
+    dest: "{{ formplayer_release_dir }}/{{ item }}"
+  with_items:
+    - application.properties
+    - sentry.properties
+  tags:
+    - formplayer_deploy
+  when: new_release_name|default(None) and current_stat.stat.exists
```

Comment:
I don't understand why we're copying these into the "current" directory, which is normally symlink? Will this `when` condition evaluate to true when `current` is a symlink? It seems like this will defeat the purpose of the atomic commit.

Reply:
My intention (which I think this does) is that this task copy files _from_ the `current` directory (if the `current` directory exists and the `formplayer_release_dir` isn't already set to the path of the `current` directory in which case we'd be copying a file to itself) to the new release directory.

This isn't strictly necessary since these files get overwritten by the following two tasks, which recreate the file contents from a template. It does however have the benefit of having the template tasks print out a diff from the existing files instead of just printing the full output with a `+` before each line.

Reply:
> This isn't strictly necessary since these files get overwritten by the following two tasks ...

Ah, yes! I saw that they were being overwritten below, and that was adding to the confusion.

> It does however have the benefit of having the template tasks print out a diff from the existing files...

That makes a lot of sense—meaningful diffs are very nice. What do you think of adding a comment to explain that the reason for this task is so the following tasks produce meaningful diffs?

Reply:
Okay yeah a comment sounds great